app/src/main/java/com/example/erabook/data/firebasedb/UserDataRemote.kt

```diff
@@ -5,12 +5,12 @@ import java.util.Date
 
 data class UserDataRemote(
     @SerializedName("userUid") val userUid: String? = null,
-    @SerializedName("userName") val userName: String? = null,
+    @SerializedName("userName") var userName: String? = null,
     @SerializedName("userEmail") val userEmail: String? = null,
 //    @SerializedName("user_location") val userLocation: Coordinates?,
-    @SerializedName("userMobile") val userMobile: Int? = null,
+    @SerializedName("userMobile") var userMobile: Int? = null,
     @SerializedName("userProfileImg") val userProfileImg: String? = null,
-    @SerializedName("userUsername") val userUsername: String? = null,
+    @SerializedName("userUsername") var userUsername: String? = null,
```

Comment:
I see you have changed back to var? Is there any specific reason?

Reply:
yes, when the user updates his info the username needs to change and be updated in the db.

Reply:
That's a bad practice, try to use the power of data class `.copy`

So you can create an object of that class and when you need to update a specific value use:
userDataRemote.copy(userUsername = "Filani")

in thsi way you will have the updated one and use it wherever you need.